### PR TITLE
feat: implement create case study API and write tests

### PR DESCRIPTION
**Pull request by Cosmoboi-i**

### CODE REVIEW

**Comment by rohini-senthil** [comment]
Let's add coverage folder to .gitignore please

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
 node_modules
 .DS_Store
 .env
-
+coverage
```
Comment (rohini-senthil) [inline .gitignore:4]: The coverage folder is still part of the file changes, let's remove that please

File: src/controllers/caseStudies.controller.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const caseStudiesServices = require('../services/caseStudies.service');
```
Comment (rohini-senthil) [inline src/controllers/caseStudies.controller.js:1]: Can we rename this file to `case-study.controller.js`

File: src/routes/caseStudies.routes.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const caseStudiesRouter = require('express').Router();
```
Comment (rohini-senthil) [inline src/routes/caseStudies.routes.js:1]: Can we rename this file to `case-study.route.js`?

File: src/controllers/caseStudies.controller.js (hunk 2 above)
Comment (rohini-senthil) [reply src/controllers/caseStudies.controller.js:1]: Similarly for the other files as well

File: database/config/config.json [hunk 4]
```diff
@@ -18,6 +18,15 @@
   "test": {
     "username": "postgres" ,
     "password": null,
+    "database": "postgres",
+    "host": "127.0.0.1",
+    "dialect": "postgres",
+    "port": 5433
+  },
```
Comment (rohini-senthil) [inline database/config/config.json:25]: I think the conflicts are resolved incorrectly here – there are two test dbs

File: .gitignore (hunk 1 above)
Comment (Cosmoboi-i) [reply .gitignore]: Done

File: src/controllers/caseStudies.controller.js (hunk 2 above)
Comment (Cosmoboi-i) [reply src/controllers/caseStudies.controller.js]: Done

File: src/routes/caseStudies.routes.js (hunk 3 above)
Comment (Cosmoboi-i) [reply src/routes/caseStudies.routes.js]: Done

File: database/config/config.json (hunk 4 above)
Comment (Cosmoboi-i) [reply database/config/config.json]: Fixed

File: src/routes/case-study.routes.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const caseStudiesRouter = require('express').Router();
```
Comment (rohini-senthil) [inline src/routes/case-study.routes.js:1]: Can we do "route" (singular) in the file name?

**Comment by rohini-senthil** [comment]
Looks good. Can you resolve conflicts with main?

**Comment by Cosmoboi-i** [comment]
Resolved

**Comment by rohini-senthil** [comment]
Can you resolve conflicts again?

**Pull request closed without merging**